runners/java-fn-execution/src/test/java/org/apache/beam/runners/fnexecution/graph/LengthPrefixUnknownCodersTest.java

```diff
@@ -144,4 +144,30 @@ public void test() throws IOException {
         CoderTranslation.fromProto(updatedCoderProto.getCoder(),
             RehydratedComponents.forComponents(updatedCoderProto.getComponents())));
   }
+
+  @Test
+  public void testIdempotent() throws IOException {
+    MessageWithComponents originalCoderProto = CoderTranslation.toProto(original);
+    Components.Builder builder = originalCoderProto.getComponents().toBuilder();
+    String coderId = LengthPrefixUnknownCoders.generateUniqueId("rootTestId",
+        originalCoderProto.getComponents().getCodersMap().keySet());
+    builder.putCoders(coderId, originalCoderProto.getCoder());
+    MessageWithComponents updatedCoderProto =
+        LengthPrefixUnknownCoders.forCoder(coderId, builder.build(), replaceWithByteArray);
+
+    Components.Builder firstDeserializedBuilder = updatedCoderProto.getComponents().toBuilder();
+    String deserializedId =
+        LengthPrefixUnknownCoders.generateUniqueId(
+            "deserializedTestId", firstDeserializedBuilder.getCodersMap().keySet());
+    builder.putCoders(deserializedId, updatedCoderProto.getCoder());
```

Comment:
Don't you mean `firstDeserializedBuilder` and not `builder`?

Reply:
no longer relevant